impeller/entity/geometry/point_field_geometry.cc

```diff
@@ -28,17 +28,17 @@ GeometryResult PointFieldGeometry::GetPositionBuffer(
     return {};
   }
 
-  Scalar min_size = 1.0f / sqrt(std::abs(determinant));
+  Scalar min_size = 0.5f / sqrt(std::abs(determinant));
   Scalar radius = std::max(radius_, min_size);
 
   HostBuffer& host_buffer = renderer.GetTransientsBuffer();
   VertexBufferBuilder<SolidFillVertexShader::PerVertexData> vtx_builder;
-
   if (round_) {
     // Get triangulation relative to {0, 0} so we can translate it to each
-    // point in turn.
+    // point in turn. Note: we intentionally used the original radius here
+    // to capture any scaling < 1.0.
```

Comment:
But the minimum size of stroked primitives is (somewhat arbitrarily chosen as) 1.0 device pixels which means you are allowing the transform to scale the size of the dots to (near) 0.

Reply:
I will change this so we enforce the min size in the tessellator logic.

Right now we're rounding too early

Reply:
The rounding was supposed to be reversible, but I think the issue here is that the rounding is 1-dimensional and the use of the rounded radius was 2-dimensional?

Also, tessellator is also used for filled shapes and those are allowed to scale to 0, so it minimally needs a flag indicating if the operation is a stroke or a fill in order to know when to apply the rounding.

Reply:
I just realized that since the rounding code in the points code is using `sqrt(determinant)`, it is probably suffering from the same issue that `GetMaxBasisXYZ` was suffering from. If the rouding in the points code uses `GetMaxBasisXY` instead, does that also solve the issue?

Reply:
hmm good point. maybe then Tessellator::FilledCircle should have separate radius and pixel_radius arguments and we can let the caller manage.

> I just realized that since the rounding code in the points code is using sqrt(determinant), it is probably suffering from the same issue that GetMaxBasisXYZ was suffering from. If the rouding in the points code uses GetMaxBasisXY instead, does that also solve the issue?

Ahh good catch

Reply:
If the new radius works, then we should just use it in the call get get the generator below.